### PR TITLE
Ingest npm data through github api #1025

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: vulnerabilities/importers/github.py [hunk 1]
```diff
@@ -208,7 +209,7 @@ def get_purl(pkg_type: str, github_name: str) -> Optional[PackageURL]:
         vendor, _, name = github_name.partition("/")
         return PackageURL(type=pkg_type, namespace=vendor, name=name)
 
-    if pkg_type in ("nuget", "pypi", "gem", "golang"):
+    if pkg_type in ("nuget", "pypi", "gem", "golang", "npm"):
```
Comment (pombredanne) [inline vulnerabilities/importers/github.py:212]: Which are the remaining ones we do not support? Do we have an issue to complete this?

File: vulnerabilities/tests/test_data/github_api/npm-expected.json [hunk 2]
```diff
@@ -0,0 +1,6175 @@
...
+  {
+    "aliases": [
+      "GHSA-3qmc-2r76-4rqp"
+    ],
+    "summary": "Redwood is vulnerable to account takeover via dbAuth \"forgot-password\" ",
+    "affected_packages": [
+      {
+        "package": {
+          "type": "npm",
+          "namespace": null,
```
Comment (pombredanne) [inline vulnerabilities/tests/test_data/github_api/npm-expected.json:1433]: The namespace is `@redwoodjs`

File: vulnerabilities/tests/test_data/github_api/npm-expected.json [hunk 3]
```diff
@@ -0,0 +1,6175 @@
...
+            "scoring_elements": ""
+          }
+        ]
+      }
+    ],
+    "date_published": "2022-11-10T15:51:01+00:00"
+  },
+  {
+    "aliases": [
+      "GHSA-3qmc-2r76-4rqp"
```
Comment (pombredanne) [inline vulnerabilities/tests/test_data/github_api/npm-expected.json:1491]: We seem to be missing the CVSS score and vector seen in https://github.com/redwoodjs/redwood/security/advisories/GHSA-3qmc-2r76-4rqp `CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:L`

File: vulnerabilities/tests/test_data/github_api/npm-expected.json (hunk 2 above)
Comment (pombredanne) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json:1433]: There is another issue that https://github.com/redwoodjs/redwood/security/advisories/GHSA-3qmc-2r76-4rqp and (https://github.com/advisories/GHSA-3qmc-2r76-4rqp and https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2022/11/GHSA-3qmc-2r76-4rqp/GHSA-3qmc-2r76-4rqp.json ) are different. Is the graphql data from GH exactly the same or not exactly the same as the JSON?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks... See a few comments inline

File: vulnerabilities/tests/test_data/github_api/npm-expected.json (hunk 2 above)
Comment (TG1999) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json:1433]: @pombredanne for the namespace issue I have already raised a PR at packageurl-python https://github.com/package-url/packageurl-python/pull/106

Comment (TG1999) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json:1433]: The GHSA reported here is GHSA-f825-f98c-gj3g and not GHSA-3qmc-2r76-4rqp.

File: vulnerabilities/tests/test_data/github_api/npm-expected.json (hunk 3 above)
Comment (TG1999) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json:1491]: Tracked in https://github.com/nexB/vulnerablecode/issues/1045

File: vulnerabilities/importers/github.py (hunk 1 above)
Comment (TG1999) [reply vulnerabilities/importers/github.py:212]: Follow up issue here https://github.com/nexB/vulnerablecode/issues/1039#issuecomment-1366458885

File: vulnerabilities/tests/test_data/github_api/npm-expected.json (hunk 2 above)
Comment (pombredanne) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json:1433]: We are going to have a data migration issue then. This should be tracked as a high priority issue and will likely need to be an improver or a data migration?
Basically we will need to fix the npm packages that are missing a namespace. But How can we even know about them?

Comment (TG1999) [reply vulnerabilities/tests/test_data/github_api/npm-expected.json]: @pombredanne bumped the version for package-url .

**Review by pombredanne — APPROVED** [review]
LGTM!
Thank you ++

**Pull request merged**